scripts/perftests/PerformanceTestUtilities.ps1

```diff
@@ -1,209 +1,466 @@
+        {
+            $localsArguments = "global-packages"
+        }
+        ElseIf ($cleanHttpCache)
+        {
+            $localsArguments = "http-cache"
+        }
+        Else
+        {
+            Log "Too risky to invoke a locals clear with the specified parameters." "yellow"
         }
-    }
-    
-    # Find the appropriate solution file for the repository. Looks for a solution file matching the repo name, 
-    # if not it takes the first available sln file in the repo. 
-    function GetSolutionFile([string]$repository,[string]$sourceDirectoryPath) {
-
-        $gitRepoName = $repository.Substring($($repository.LastIndexOf('/') + 1))
-        $potentialSolutionFile = [System.IO.Path]::Combine($sourceDirectoryPath, "$($gitRepoName.Substring(0, $gitRepoName.Length - 4)).sln")
 
-        if(Test-Path $potentialSolutionFile)
+        If ($isClientDotnetExe)
+        {
+            . $nugetClientFilePath nuget locals -c $localsArguments *>>$null
+        }
+        Else
         {
-            $solutionFile = $potentialSolutionFile
-        } 
-        else 
+            . $nugetClientFilePath locals -clear $localsArguments -Verbosity quiet
+        }
+
+        If ($isPackagesConfig -And ($cleanGlobalPackagesFolder -Or $cleanHttpCache))
         {
-            $possibleSln = Get-ChildItem $sourceDirectoryPath *.sln
-            if($possibleSln.Length -eq 0)
-            {
-                Log "No solution files found in $sourceDirectoryPath" "red"
-            } 
-            else 
-            {
-            $solutionFile = $possibleSln[0] | Select-Object -f 1 | Select-Object -ExpandProperty FullName
-            }
+            Remove-Item $solutionPackagesFolderPath -Recurse -Force -ErrorAction Ignore > $Null
+            mkdir $solutionPackagesFolderPath > $Null
         }
-        return $solutionFile;
     }
 
-    # Given a repository and a hash, checks out the revision in the given source directory. The return is a solution file if found. 
-    function SetupGitRepository([string]$repository, [string]$commitHash, [string]$sourceDirectoryPath)
+    if($killMsBuildAndDotnetExeProcesses)
     {
-        Log "Setting up $repository into $sourceDirectoryPath"
-        DownloadRepository $repository $commitHash $sourceDirectoryPath
-        $solutionFile = GetSolutionFile $repository $sourceDirectoryPath
-        Log "Completed the repository setup. The solution file is $solutionFile" -color "Green"
-        return $solutionFile
+        Stop-Process -name msbuild*,dotnet* -Force
     }
 
-    # runs locals clear all with the given client
-    function LocalsClearAll([string]$nugetClient)
+    $arguments = [System.Collections.Generic.List[string]]::new()
+
+    $arguments.Add("restore")
+    $arguments.Add($solutionFilePath)
+
+    If ($isPackagesConfig)
     {
-        $nugetClient = GetAbsolutePath $nugetClient
-        if($(IsClientDotnetExe $nugetClient))
+        If ($isClientDotnetExe)
```

Comment:
dotnet.exe cannot do packages.config restores. I'd skip it and print a warning message.